### DISCUSSION TITLE
Spotify as a tag source

### DISCUSSION THREAD

**Discussion by arsaboo** [MoeMusic/Moe#192] [Ideas]
This is a great start already. I would love to see Spotify as another tag source. In addition to the tags, Spotify can also provide popularity and track attributes that can be extremely useful in creating smart playlists. [Spotipy](https://github.com/plamere/spotipy) is a great library that already has all the endpoints.

**Comment by jtpavlock** [comment]
This is a great idea, but I don't actually think I'll implement this or support this in Moe. Let me explain.

To me, this is a perfect idea for a third-party plugin. I'd like to keep the core repository of Moe relatively limited, while creating a system to make plugin development as seamless as possible. I believe this will help Moe in the long run allowing for easier maintenance and development of other core features.

I added musicbrainz to Moe as I think it makes sense to at least ship with one metadata-source, but at the moment I don't plan on adding any more. That isn't to say I won't develop my own plugins for other sources (personally I'm eagerly awaiting the sonemic api), but when that happens I plan on creating them as external plugins in their own repositories.

It shouldn't be too difficult at all to implement spotify as another tag source, and I'd be happy to help you create your own plugin to do so if that's something you're interested in.

**Comment by arsaboo** [comment]
I am 100% in agreement with that view. I worked on a couple of Beets plugins (including Spotify) and have an open [PR](https://github.com/beetbox/beets/pull/4418) to add Plex support. I would love to add these plugins here. Once I see a few more third-party plugins, I will give it a shot (unless someone else beats me to it).
